index.js

```diff
@@ -1,7 +1,7 @@
 import process from 'node:process';
 import {userInfo} from 'node:os';
 
-const defaultShell = (() => {
+export const detectShell = () => {
```

Comment:
```suggestion
export const detectDefaultShell = () => {
```

This makes the naming more consistent.